Source/MSBuildTools.Unity/Packages/com.microsoft.msbuildforunity/Editor/ProjectGenerator/Scripts/MSBuildTools.cs

```diff
@@ -31,21 +65,48 @@ public static class MSBuildTools
         };
 
         public const string CSharpVersion = "7.3";
+        public const string AutoGenerate = "MSBuild/Generation Enabled";
+
         public static readonly Version DefaultMinUWPSDK = new Version("10.0.14393.0");
 
-        private static readonly string uwpMinPlatformVersion = EditorUserBuildSettings.wsaMinUWPSDK;
-        private static readonly string uwpTargetPlatformVersion = EditorUserBuildSettings.wsaUWPSDK;
+        private static UnityProjectInfo unityProjectInfo;
+
+        public static UnityProjectInfo UnityProjectInfo => unityProjectInfo ?? (unityProjectInfo = new UnityProjectInfo(supportedBuildTargets));
 
         private static IProjectExporter exporter = null;
 
-        private static IProjectExporter Exporter => exporter ?? (exporter = new TemplatedProjectExporter(new DirectoryInfo(Utilities.MSBuildProjectFolder), TemplateFiles.Instance.MSBuildSolutionTemplatePath, TemplateFiles.Instance.SDKProjectFileTemplatePath, TemplateFiles.Instance.SDKProjectPropsFileTemplatePath, TemplateFiles.Instance.SDKProjectTargetsFileTemplatePath));
+        private static IProjectExporter Exporter => exporter ?? (exporter = new TemplatedProjectExporter(new DirectoryInfo(Utilities.MSBuildProjectFolder),
+            TemplateFiles.Instance.MSBuildSolutionTemplatePath,
+            TemplateFiles.Instance.SDKProjectFileTemplatePath,
+            TemplateFiles.Instance.SDKGeneratedProjectFileTemplatePath,
+            TemplateFiles.Instance.SDKProjectPropsFileTemplatePath,
+            TemplateFiles.Instance.SDKProjectTargetsFileTemplatePath,
+            TemplateFiles.Instance.MSBuildForUnityCommonPropsTemplatePath));
+
+        public static MSBuildToolsConfig Config { get; } = MSBuildToolsConfig.Load();
+        
+        [MenuItem(AutoGenerate, priority = 101)]
```

Comment:
Should these go under the `MSBuild` menu?

Reply:
They do :)

Reply:
Nevermind! I see now that this is a variable with the full path defined